libraries/SITL/SIM_SilentWings.cpp

```diff
@@ -0,0 +1,299 @@
+    { "INS_ACCOFFS_Y",     0.001 },
+    { "INS_ACCOFFS_Z",     0.001 },
+    { "INS_ACCSCAL_X",     1.001 },
+    { "INS_ACCSCAL_Y",     1.001 },
+    { "INS_ACCSCAL_Z",     1.001 },
+};
+
+SilentWings::SilentWings(const char *home_str, const char *frame_str) :
+    Aircraft(home_str, frame_str),
+    last_data_time_ms(0),
+    first_pkt_timestamp_ms(0),
+    time_base_us(0),
+    sock(true),
+    home_initialized(false),
+    inited_first_pkt_timestamp(false)
+{
+    // Try to bind to a specific port so that if we restart ArduPilot
+    // SilentWings keeps sending us packets. Not strictly necessary but
+    // useful for debugging
+    sock.bind("127.0.0.1", 6060);
+
+    sock.reuseaddress();
+    sock.set_blocking(false);
+    
+    // Force ArduPlane to use sensor data from SilentWings as the actual state,
+    // without using EKF, i.e., using "fake EKF (type 10)". Disable gyro calibration.
+    // Set a few other parameters to specific values to keep the calibration checks happy.
+	// TO DO: fix this. Setting parameters in this way doesn't appear to have any effect.
+    for (uint8_t i = 0; i < ARRAY_SIZE(sim_defaults); i++) {
+        AP_Param::set_default_by_name(sim_defaults[i].name, sim_defaults[i].value);
+        if (sim_defaults[i].save) {
+            enum ap_var_type ptype;
+            AP_Param *p = AP_Param::find(sim_defaults[i].name, &ptype);
+            if (!p->configured()) {
+                p->save();
+            }
+        }
+    }
+}
+
+/*
+  decode and send servos
+*/
+void SilentWings::send_servos(const struct sitl_input &input)
+{
+    char *buf = nullptr;
+    // Turn off direct joystick input to the simulator. All joystick commands
+    // should go through Mission Planner and get properly fused with ArduPlane's
+    // control inputs when in automatic flight modes.
+    float joystick = 0.0f;
+    float aileron  = filtered_servo_angle(input, 0);
+    float elevator = filtered_servo_angle(input, 1);
+    float throttle = filtered_servo_range(input, 2);
+    float rudder   = filtered_servo_angle(input, 3);
+    float flap     = 0.0f;
+
+    asprintf(&buf,
+             "JOY %f\n"
+             "AIL %f\n"
+             "ELE %f\n"
+             "RUD %f\n"
+             "THR %f\n"
+             "FLP %f\n",
+             joystick, aileron, elevator, rudder, throttle, flap);
+
+    ssize_t buflen = strlen(buf);
+    
+    ssize_t sent = sock.sendto(buf, buflen, "127.0.0.1", 6070);
+    free(buf);
+
+    if (sent < 0) {
+        fprintf(stderr, "Fatal: Failed to send on control socket\n"),
+        exit(1);
+    }
+    
+    if (sent < buflen) {
+        fprintf(stderr, "Failed to send all bytes on control socket\n");
+    }
+}
+
+
+/*
+  Receive an update from the FDM
+  This is a blocking function
+ */
+bool SilentWings::recv_fdm(void)
+{
+    fdm_packet tmp_pkt;
+    memset(&pkt, 0, sizeof(pkt));
+
+    ssize_t nread = sock.recv(&tmp_pkt, sizeof(pkt), 0);
+    
+    // nread == -1 (255) means no data has arrived
```

Comment:
comment says it's 255 which is wrong. ssize_t is signed size_t so it's actually -1.

Reply:
Yes, but if someone decides to print this value an unsigned, they'll see 255; that's what the comment is referring to.